<?xml version="1.0" ?>
<robot name="server" xmlns:xacro="http://ros.org/wiki/xacro">
  <xacro:arg name="botname" default=""/>
  <xacro:include filename="$(find multi_camera)/urdf/cameras.gazebo.xacro"/>
  <xacro:include filename="$(find turtlebot3_description)/urdf/common_properties.xacro"/>

  <link name="base_footprint"/>

  <joint name="base_joint" type="fixed">
    <parent link="base_footprint"/>
    <child link="camera1_mount"/>
    <origin xyz="0.0 0.0 0.010" rpy="0 0 0"/>
  </joint>

  <link name="camera1_mount">
    <visual>
      <origin xyz="-0.032 0 0.0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://multi_camera/models/mount.dae" scale="1 1 1"/>
      </geometry>
      <material name="brown"/>
    </visual>

    <collision>
      <origin xyz="-0.032 0 0.070" rpy="0 0 0"/>
      <geometry>
        <box size="0.140 0.140 0.143"/>
      </geometry>
    </collision>

    <inertial>
      <origin xyz="0 0 0 " rpy="0 0 0"/>
      <mass value="8.2573504e-01"/>
      <inertia ixx="2.2124416e-03" ixy="-1.2294101e-05" ixz="3.4938785e-05"
               iyy="2.1193702e-03" iyz="-5.0120904e-06"
               izz="2.0064271e-03" />
    </inertial>
  </link>

<link name="camera1_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.0634432857550415"/>
      <inertia ixx="1.68383298504221E-05" ixy="-1.0448046286551E-07" ixz="4.28825686069912E-07" 
               iyy="2.10729825490332E-05" iyz="-9.43318165957986E-09" 
               izz="3.56809250260432E-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://multi_camera/models/camera.dae"/>
      </geometry>
      <material name="gray">
        <color rgba="0.250980392156863 0.250980392156863 0.250980392156863 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://multi_camera/models/camera.dae"/>
      </geometry>
    </collision>
  </link>
  
  <joint name="camera1_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0.98"/>
    <parent link="camera1_mount"/>
    <child link="camera1_link"/>
  </joint>

  
</robot>
